### PR TITLE
refactor: @snyk/fix

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

Refactor the @snyk/fix code to split helper functions into separate files for readability & include tests for these.
Rename some variable to clarify that these are internal concepts.

### CODE REVIEW

File: packages/snyk-fix/src/index.ts [hunk 1]
```diff
@@ -92,3 +88,19 @@ export function groupEntitiesPerScanType(
   }
   return entitiesPerType;
 }
+
+export function extractMeta(
+  resultsByPlugin: FixHandlerResultByPlugin,
+  exceptionsByScanType: ErrorsByEcoSystem,
+): {
+  fixed: number;
```
Comment (aarlaud) [inline packages/snyk-fix/src/index.ts:96]: shouldn't this return be a type?

Comment (lili2311) [reply packages/snyk-fix/src/index.ts]: updated

File: packages/snyk-fix/test/unit/fix.spec.ts [hunk 2]
```diff
@@ -172,7 +172,7 @@ describe('Snyk fix', () => {
     ).toEqual('prod.txt');
   });
 
-  it('Snyk fix returns results as expected when remediation data is empty', async () => {
+  it.only('Snyk fix returns results as expected when remediation data is empty', async () => {
```
Comment (lili2311) [inline packages/snyk-fix/test/unit/fix.spec.ts:175]: removing this + 👀 why this test is failing

Comment (lili2311) [reply packages/snyk-fix/test/unit/fix.spec.ts]: solved

**Review by aarlaud — APPROVED** [review]

**Pull request merged**